docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1938 @@
-this policy also allows egress UDP traffic on port 53 to all Pods with the
-label `app: dns`. Similar to K8s NetworkPolicy, Antrea will only install OVS
-flows for this ACNP on Nodes for which some of the Pods are the target of the
-policy. Thus, we have scheduled three Pods (appServer, appDns, appNotClient)
-on the same Node and they have the following IP addresses:
+This ACNP is applied to all Pods with the label `app: web` in all Namespaces. For these Pods, only ingress HTTP
+traffic with method `GET` path `/api/v2/*` from the Pods with the label `app: client` is allowed, and others will
+be dropped.
 
-- appServer: 10.10.1.6
-- appNotClient: 10.10.1.7
-- appDns: 10.10.1.8
+```yaml
+apiVersion: crd.antrea.io/v1beta1
+kind: NetworkPolicy
+metadata:
+  name: ingress-allow-http-request-to-api-v2
+spec:
+  priority: 6
+  tier: application
+  appliedTo:
+    - podSelector:
+        matchLabels:
+          app: web
+  ingress:
+    - name: AllowFromClientL7
+      action: Allow
+      from:
+        - podSelector:
+            matchLabels:
+              app: client
+      l7Protocols:
+        - http:
+            path: "/api/v2/*"
+            method: "GET"
+    - action: Drop
+```
 
-## Tables
+## TrafficControl Implementation
 
-![OVS pipeline](../assets/ovs-pipeline-antrea-proxy.svg)
+[TrafficControl](../traffic-control.md) is a CRD API that manages and manipulates the transmission of Pod traffic.
+Antrea creates a dedicated table [TrafficControl] to implement `TrafficControl`. Use the following TrafficControls
+as examples throughout the remainder of this document.
 
-### ClassifierTable (0)
+This is a TrafficControl applied to Pods with the label `app: web`. For these Pods, both ingress and egress traffic of
+them will be redirected to the port `antrea-tc-tap0`, and returned back via the port `antrea-tc-tap1`.
 
-This table is used to determine which "category" of traffic (tunnel, local
-gateway or local Pod) the packet belongs to. This is done by matching on the
-ingress port for the packet. The appropriate value is then written to bits
-[0..3] in NXM_NX_REG0: 0 for tunnel, 1 for local gateway and 2 for local Pod.
-This information is used by matches in subsequent tables. For a packet received
-from the tunnel port, bit 19 in NXM_NX_REG0 is set to 1, to indicate MAC rewrite
-should be performed for the packet in [L3ForwardingTable].
+```yaml
+apiVersion: crd.antrea.io/v1alpha2
+kind: TrafficControl
+metadata:
+  name: redirect-web-to-local
+spec:
+  appliedTo:
+    podSelector:
+      matchLabels:
+        app: web
+  direction: Both
+  action: Redirect
+  targetPort:
+    ovsInternal:
+      name: antrea-tc-tap0
+  returnPort:
+    ovsInternal:
+      name: antrea-tc-tap1
+```
 
-If you dump the flows for this table, you may see the following:
+This is a TrafficControl applied to Pods with the label `app: db`. For these Pods, all traffic, both ingress and egress,
+will be mirrored to the port `antrea-tc-tap2`.
```

Comment:
For these Pods, both ingress and egress traffic will be mirrored (duplicated) to port `antrea-tc-tap2`.